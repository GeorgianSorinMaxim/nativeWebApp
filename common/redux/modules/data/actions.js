// @flow

export const APP_LAUNCHED = 'data/APP_LAUNCHED';

export const START_FETCH_DATA = 'data/START_FETCH';

export const DATA_FETCH_SUCCESS = 'data/FETCH_SUCCESS';
export const DATA_FETCH_FAILURE = 'data/FETCH_FAILURE';

type AppLaunched = {
  type: 'data/APP_LAUNCHED'
};

type StartDataFetch = {
  type: 'data/START_FETCH'
};

type DataFetchSuccess = {
  type: 'data/FETCH_SUCCESS',
  payload: Object
};

type DataFetchFailure = {
  type: 'data/FETCH_FAILURE',
  payload: Object
};

export function appLaunched(): AppLaunched {
  return { type: APP_LAUNCHED };
}

export function startFetchingData(): StartDataFetch {
  return { type: START_FETCH_DATA };
}

export function dataFetchSuccess(payload: Object): DataFetchSuccess {
  return { type: DATA_FETCH_SUCCESS, payload };
}

export function dataFetchFailure(error: Object): DataFetchFailure {
  return { type: DATA_FETCH_FAILURE, payload: { error } };
}

export type Action =
  | AppLaunched
  | StartDataFetch
  | DataFetchSuccess
  | DataFetchFailure;
