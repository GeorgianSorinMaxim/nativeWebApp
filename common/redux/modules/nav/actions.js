// @flow
import { NavigationActions } from 'react-navigation';

const navToPath = path => NavigationActions.navigate({ routeName: path });
// const navToPathWithParams = (path: string, params: Object) =>
//   NavigationActions.navigate({ routeName: path, params });

const navToHome = () => navToPath('Home');
const navToFeature = () => navToPath('Feature');

export const actions = {
  navToHome,
  navToFeature
};
