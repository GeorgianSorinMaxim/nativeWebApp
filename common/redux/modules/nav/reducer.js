// @flow
import { AppNavigator } from '../../../navigator/AppNavigator';

export type State = {
  index: number,
  routes: Array<any>
};

export const initialState: ?State = AppNavigator.router.getStateForAction(
  AppNavigator.router.getActionForPathAndParams('App')
);

export default function reducer(
  state: State = initialState,
  action: Action
): State {
  const nextState = AppNavigator.router.getStateForAction(action, state);
  return nextState || state;
}
