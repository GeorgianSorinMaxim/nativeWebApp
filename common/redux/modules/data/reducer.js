// @flow

import * as actions from './actions';

import type { Action } from './actions';

export type State = {
  entities: Object,
  loading: boolean,
  error?: any
};

export const initialState: State = {
  entities: {},
  loading: false
};

export default function reducer(
  state: State = initialState,
  action: Action
): State {
  switch (action.type) {
    case actions.START_FETCH_DATA:
      return {
        ...state,
        loading: true
      };

    case actions.DATA_FETCH_SUCCESS:
      return {
        ...state,
        entities: action.payload,
        loading: false
      };

    case actions.DATA_FETCH_FAILURE:
      return {
        ...state,
        error: action.payload,
        loading: false
      };

    default:
      return state;
  }
}
