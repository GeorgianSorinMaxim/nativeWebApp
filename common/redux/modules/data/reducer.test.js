// @flow

import reducer from './reducer';
import * as actions from './actions';
import type { State } from './reducer';

const initialState: State = {
  entities: {},
  loading: false
};

describe('Reducer', () => {
  it('should start fetching data', () => {
    const state = reducer(initialState, { type: 'data/START_FETCH' });
    expect(state).toMatchSnapshot();
  });

  it('should finish fetching data sucessfully', () => {
    const action = actions.dataFetchSuccess({});
    const state = reducer(initialState, action);
    expect(state).toMatchSnapshot();
  });

  it('should finish fetching data with a failure', () => {
    const action = actions.dataFetchFailure({});
    const state = reducer(initialState, action);
    expect(state).toMatchSnapshot();
  });
});
