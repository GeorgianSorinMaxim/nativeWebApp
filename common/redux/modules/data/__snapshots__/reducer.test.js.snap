// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reducer should finish fetching data sucessfully 1`] = `
Object {
  "entities": Object {},
  "loading": false,
}
`;

exports[`Reducer should finish fetching data with a failure 1`] = `
Object {
  "entities": Object {},
  "error": Object {
    "error": Object {},
  },
  "loading": false,
}
`;

exports[`Reducer should start fetching data 1`] = `
Object {
  "entities": Object {},
  "loading": true,
}
`;
